src/containers/login/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { useSelector, useDispatch } from '../../store/index'; 
+import {loginUser} from "../../store/actions/login";
+import { useNavigate } from "react-router-dom";
+import profile from "..//../component/image/undraw_login_re_4vu2.svg";
+
+import '..//../styles/sass/main.scss'
+
+
+
+export const LoginPage = () => {
+  const {error,userLoggedIn} = useSelector((state) => state.user)
+  const dispatch = useDispatch()
+  const [username, setUsername] = useState("");
+  const [personalAccessToked, setPersonalAccessToked] = useState("");
+  const [usernameMissing, setUsernameMissing] = useState(false);
+  const [personalAccessTokenMissing, setPersonalAccessTokenMissing] = useState(false);
+  const [validateUserData, setvalidateUserData] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const navigate = useNavigate();
+  useEffect(()=>{
+    if(userLoggedIn){
+      navigate("/");
+    }
+  },[userLoggedIn,navigate])
+
+  useEffect(()=>{
+    if(error !== ''&&loading==true){
+      setvalidateUserData(true);
+      setUsernameMissing(false);
+      setPersonalAccessTokenMissing(false);
+    }
+  },[error,loading])
+
+  function login() {
+    if(username===''&&personalAccessToked===''){
+      setUsernameMissing(true);
+      setPersonalAccessTokenMissing(true);
+      setvalidateUserData(false);
+      return;
+    }
+    if(username===''){
+      setUsernameMissing(true);
+      setPersonalAccessTokenMissing(false);
+      setvalidateUserData(false);
+      return;
+    }
+    if(personalAccessToked===''){
+      setUsernameMissing(false);
+      setPersonalAccessTokenMissing(true);
+      setvalidateUserData(false);
+      return;
+    }
+    if(username===''||personalAccessToked===''){
+      return;
+    }
+    dispatch(loginUser(personalAccessToked,username));
+  }
+  function setLoader(){
+      setLoading(true);
+    setTimeout(() => {
+      setLoading(false);
+      login()
+    }, 100);
+  };
+
+  return(
+    <div>
+      <div className="login">
+        <div className="sub-login">
+        <div className="imgs">
+          <div className="contanier-image">
+            <img src={profile} alt="profile" className="profile"/>
+          </div>
+        </div>
+        <div className="right">
+          <div className="first-input">
+            <input className="input-feild name" value={username} onChange={(e) => setUsername(e.target.value)} type="text" placeholder="username" />
+          </div>
+          {usernameMissing &&  <p className='errors'>Please enter your username</p>}
+          <div className="second-input">
+            <input className="input-feild name" value={personalAccessToked} onChange={(e) => setPersonalAccessToked(e.target.value)} type="password" placeholder="personalAccessToked" />
+            {personalAccessTokenMissing &&  <p className='errors'>Please enter your personal Access Token</p>}
+            {validateUserData &&  <p className='errors'>{error}</p>}
```

Comment:
No need to maintain `validateUserData` state for showing the redux error state, you can just use `error` to check if the error exists and show it, like below
```suggestion
            {error &&  <p className='errors'>{error}</p>}
```
Now you don't need to maintain a local state for showing the error. 
Always try to use the global state directly without storing it in the local state variables otherwise you'll have to write few extra lines of code to sync the local state with the global state, and if not done properly, it'll lead to inconsistencies and can introduce bugs later on.

Reply:
`validateUserData` should be there, let say if we have removed that and then we added wrong username and password , it will give `Either username or personal Access Token is wrong` and if we again reload the page then again it will give that error